submissions/models.py

```diff
@@ -33,6 +33,11 @@
     providing_args=['anonymous_user_id', 'course_id', 'item_id']
 )
 
+# In order to allow for effective soft-deletes, we want to override the default Manager for several object types
+class SoftDeletedManager(models.Manager):
+    def get_queryset(self):
+        #By default, filter out any soft-deleted records for classes using this manager
+        return super(SoftDeletedManager, self).get_queryset().filter(deleted=False)
```

Comment:
Thanks @mulby, that's very helpful.

I found the PR where @cahrens is working on the data downloads here(https://github.com/edx/edx-platform/pull/11167), and it's also using SQL instead of models, so I think we'll be covered on all fronts.